test/integration/crud/explain.test.ts

```diff
@@ -80,203 +69,182 @@ describe('Explain', function () {
+      } else if (aggResult.$cursor) {
+        expect(aggResult.$cursor).to.have.property('queryPlanner');
+        expect(aggResult.$cursor).to.not.have.property('executionStats');
+      } else {
+        expect(aggResult).to.have.property('queryPlanner');
+        expect(aggResult).to.not.have.property('executionStats');
+      }
+    });
 
-    await collection.insertOne({ a: 1 });
-    const aggResult = await collection
-      .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
-      .explain();
-    if (aggResult && aggResult.stages) {
-      expect(aggResult.stages).to.have.length.gte(1);
-      expect(aggResult.stages[0]).to.have.property('$cursor');
-      expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
-      expect(aggResult.stages[0].$cursor).to.have.property('executionStats');
-    } else {
-      expect(aggResult).to.have.property('queryPlanner');
-      expect(aggResult).to.have.property('executionStats');
-    }
+    it('when explain is set to "allPlansExecution", aggregate result returns queryPlanner and executionStats properties', async function () {
+      const aggResult = await collection
+        .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
+        .explain('allPlansExecution');
+
+      if (aggResult && aggResult.stages) {
+        expect(aggResult.stages).to.have.length.gte(1);
+        expect(aggResult.stages[0]).to.have.property('$cursor');
+        expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
+        expect(aggResult.stages[0].$cursor).to.have.property('executionStats');
+      } else {
+        expect(aggResult).to.have.property('queryPlanner');
+        expect(aggResult).to.have.property('executionStats');
+      }
+    });
+
+    it('when explain is not set, aggregate result returns queryPlanner and executionStats properties', async function () {
+      const aggResult = await collection
+        .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }])
+        .explain();
+      if (aggResult && aggResult.stages) {
+        expect(aggResult.stages).to.have.length.gte(1);
+        expect(aggResult.stages[0]).to.have.property('$cursor');
+        expect(aggResult.stages[0].$cursor).to.have.property('queryPlanner');
+        expect(aggResult.stages[0].$cursor).to.have.property('executionStats');
+      } else {
+        expect(aggResult).to.have.property('queryPlanner');
+        expect(aggResult).to.have.property('executionStats');
+      }
+    });
   });
 
-  it('should throw a catchable error with invalid explain string', async function () {
-    const db = client.db('shouldThrowCatchableError');
-    const collection = db.collection('test');
-    const error = await collection
-      .find({ a: 1 })
-      .explain('invalidExplain')
-      .catch(error => error);
-    expect(error).to.be.instanceOf(MongoServerError);
+  context('when explain is set to "invalidExplain", result returns MongoServerError', () => {
```

Comment:
In this context, we don't necessarily care that explain is set to `'invalidExplain'` exactly. More so, we just want to know that when the value provided isn't something we expect, i.e. it's not `true`, `false`, `'allPlansExecution'`, `'queryPlanner'` or `'executionStats'`, so we can say that explain is set to 'an invalid value'.

Also, when using context blocks, we use them to describe exactly that: the context. We leave what the expectation is for our `it` blocks.

Think of reading the test like this, first you have your top level `describe` block and then you have your `context` block describing the scenario (which can be nested) and then you have the `it` block describing the expected outcome. Taking a look at what this would look like here, we'd have

> [describe] Explain: [context] when explain is set to an invalid value, result returns MongoServerError, [it] should throw a catchable error with invalid explain string

which is clunky to read as it duplicates information.

We want it to read more like:

> [describe] Explain: [context] when explain is set to an invalid value, [it] should throw a catchable error with invalid explain string